test/README.md

```diff
@@ -0,0 +1,149 @@
+# Habitat testing
+
+## Running all tests
+
+First, ensure that all Habitat related environment variables are unset:
```

Comment:
I originally started implementing the test stuff so it could run completely safely in any environment, but there were issues with:

- tl;dr - making the test suite use it's own safe env was extremely complicated
- managing a separate cache of keys inside and outside of a studio
  - setting the `HAB_CACHE_KEY_PATH` env var isn't passed on through to a studio, and the paths wouldn't be the same inside and outside of the studio anyways.
- trying to set alt `HAB_STUDIOS_HOME`, `HAB_STUDIO_ROOT`, `HAB_ROOT` didn't provide the desired results.

we talked a bit on Slack about making a separate Makefile target such as `make test-shell`, but in the end, it seemed like `make shell` got us what we needed.